recipes/llama-cpp/all/conanfile.py

```diff
@@ -114,23 +99,80 @@ def package(self):
         copy(self, "*common*.a", src=self.build_folder, dst=os.path.join(self.package_folder, "lib"), keep_path=False)
         copy(self, "*.cmake", src=os.path.join(self.export_sources_folder, "cmake"), dst=os.path.join(self.package_folder, "lib", "cmake"))
 
-    def package_info(self):
-        self.cpp_info.components["common"].includedirs = [os.path.join("include", "common")]
-        self.cpp_info.components["common"].libs = ["common"]
-        self.cpp_info.components["common"].libdirs = ["lib"]
-        if self.version >= Version("b3240"):
-            self.cpp_info.components["common"].libs.append("ggml")
-
-        self.cpp_info.components["llama"].libs = ["llama"]
-        self.cpp_info.components["llama"].resdirs = ["res"]
-        self.cpp_info.components["llama"].libdirs = ["lib"]
-
-        if self.options.with_cuda and not self.options.shared:
-            self.cpp_info.builddirs.append(os.path.join("lib", "cmake"))
-            module_path = os.path.join("lib", "cmake", "llama-cpp-cuda-static.cmake")
-            self.cpp_info.set_property("cmake_build_modules", [module_path])
-
+    def _get_default_backends(self):
+        results = ["cpu"]
         if is_apple_os(self):
-            self.cpp_info.components["common"].frameworks.extend(["Foundation", "Accelerate", "Metal"])
-        elif self.settings.os in ("Linux", "FreeBSD"):
-            self.cpp_info.components["common"].system_libs.extend(["dl", "m", "pthread", "gomp"])
+            results.append("blas")
+            results.append("metal")
+        return results
+
+    def package_info(self):
+        # New structure in llama
+        if Version(self.version) >= "b4570":
+            self.cpp_info.components["common"].includedirs = [os.path.join("include", "common")]
+            self.cpp_info.components["common"].libs = ["common"]
+            self.cpp_info.components["common"].requires = ["llama"]
```

Comment:
Is this true? I'm not sure that llama requires common, this requires was not defined for previous versions

Reply:
`target_link_libraries     (${TARGET} PRIVATE ${LLAMA_COMMON_EXTRA_LIBS} PUBLIC llama Threads::Threads)` with `TARGET` being `common` here too

Reply:
Seems like it was missing for the old versions too, as this same line is present in the previous version too